### PR TITLE
Make MemberList nothrow move constructible.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/ast/Types.h [hunk 1]
```diff
@@ -132,6 +131,8 @@ class MemberList
 	mutable std::unique_ptr<StorageOffsets> m_storageOffsets;
 };
 
+static_assert(std::is_nothrow_move_constructible<MemberList>::value, "MyType should be noexcept MoveConstructible");
```
Comment (axic) [inline libsolidity/ast/Types.h]: MyType?

Comment (chriseth) [reply libsolidity/ast/Types.h]: Sorry, copypasta

**Comment by chriseth** [comment]
Updated.

**Comment by axic** [comment]
All the `is_nothrow_move_constructible` documentation showed was constructors and not the assignment operator. Is that covered by this?

**Comment by chriseth** [comment]
Yes, the assignment operator prevented the proper move constructor to be generated. I also remove the empty default constructor.

**Comment by chriseth** [comment]
Actually I thought this class would be used in vectors but it turns out it is not, so this change is not such a big deal.

**Review by ekpyron — APPROVED** [review]

**Pull request merged**